src/Lexer.php

```diff
@@ -394,33 +443,36 @@ protected function scanEOS()
     }
 
     /**
-     * Returns keywords for provided type.
+     * Returns a regex matching the keywords for the provided type.
      *
      * @param string $type Keyword type
      *
      * @return string
+     *
+     * @deprecated
      */
     protected function getKeywords($type)
     {
-        if (!isset($this->keywordsCache[$type])) {
-            $getter = 'get' . $type . 'Keywords';
-            $keywords = $this->keywords->$getter();
-
-            if ($type === 'Step') {
-                $padded = [];
-                foreach (explode('|', $keywords) as $keyword) {
-                    $padded[] = str_contains($keyword, '<')
-                        ? preg_quote(mb_substr($keyword, 0, -1, 'utf8'), '/') . '\s*'
-                        : preg_quote($keyword, '/') . '\s+';
```

Comment:
If I've understood right, `currentDialect->getStepKeywords()` will return the list of keywords, which may or may not have a trailing space, but that's not the same as the current behaviour where the return from this method matches *at least* zero-or-one spaces depending on how the keyword is defined?

Reply:
the "at least" is covered by the fact that I'm left-trimming the step text.

Reply:
I might be misunderstanding, but isn't this the "dead code" for BC with classes that have extended the Lexer with e.g. a custom `scanStep` that is still using the old `scanInputForKeywords` though? So they will still need the regex as before?